packages/pubsub/src/subscription.js

```diff
@@ -592,30 +350,154 @@ Subscription.prototype.decorateMessage_ = function(message) {
   var self = this;
 
-  callback = callback || common.util.noop;
+  if (is.fn(gaxOpts)) {
+    callback = gaxOpts;
+    gaxOpts = {};
+  }
 
-  var protoOpts = {
-    service: 'Subscriber',
-    method: 'deleteSubscription'
+  var reqOpts = {
+    subscription: this.name
   };
 
+  this.request({
+    client: 'subscriberClient',
+    method: 'deleteSubscription',
+    reqOpts: reqOpts,
+    gaxOpts: gaxOpts
+  }, function(err, resp) {
+    if (!err) {
+      self.closeConnection_();
+    }
+
+    callback(err, resp);
+  });
+};
+
+/**
+ * @param {function} callback - The callback function.
+ * @param {?error} callback.err - An error returned while making this
+ *     request.
+ * @param {object} callback.apiResponse - Raw API response.
+ */
+Subscription.prototype.getMetadata = function(gaxOpts, callback) {
+  if (is.fn(gaxOpts)) {
+    callback = gaxOpts;
+    gaxOpts = {};
+  }
+
   var reqOpts = {
     subscription: this.name
   };
 
-  this.parent.request(protoOpts, reqOpts, function(err, resp) {
+  this.request({
+    client: 'subscriberClient',
+    method: 'getSubscription',
+    reqOpts: reqOpts,
+    gaxOpts: gaxOpts
+  }, callback);
+};
+
+/**
+ * Begin listening for events on the subscription. This method keeps track of
+ * how many message listeners are assigned, and then removed, making sure
+ * polling is handled automatically.
+ *
+ * As long as there is one active message listener, the connection is open. As
+ * soon as there are no more message listeners, the connection is closed.
+ *
+ * @private
+ *
+ * @example
+ * subscription.listenForEvents_();
+ */
+Subscription.prototype.listenForEvents_ = function() {
+  var self = this;
+
+  this.on('newListener', function(event) {
+    if (event === 'message') {
+      self.messageListeners++;
+
+      if (!self.connection) {
+        self.openConnection_();
+      }
+    }
+  });
+
+  this.on('removeListener', function(event) {
+    if (event === 'message' && --self.messageListeners === 0) {
+      self.closeConnection_();
+    }
+  });
+};
+
+/**
+ * @param {object} config - The push config.
+ * @param {string} config.pushEndpoint
+ * @param {object} config.attributes
+ */
+Subscription.prototype.modifyPushConfig = function(config, gaxOpts, callback) {
+  if (is.fn(gaxOpts)) {
+    callback = gaxOpts;
+    gaxOpts = {};
+  }
+
+  var reqOpts = {
+    subscription: this.name,
+    pushConfig: config
+  };
+
+  this.request({
+    client: 'subscriberClient',
+    method: 'modifyPushConfig',
+    reqOpts: reqOpts,
+    gaxOpts: gaxOpts
+  }, callback);
+};
+
+/**
+ * @private
+ */
+Subscription.prototype.openConnection_ = function() {
+  var self = this;
+
+  var reqOpts = {
+    subscription: this.name,
+    streamAckDeadlineSeconds: this.ackDeadline
+  };
+
+  this.request({
+    client: 'subscriberClient',
+    method: 'streamingPull',
+    returnFn: true
+  }, function(err, requestFn) {
     if (err) {
-      callback(err, resp);
+      self.emit('error', err);
       return;
```

Comment:
GAX should handle all the retrying under the hood, otherwise yes, we pass the error back to the user by emitting an error event.